examples/api/mongocxx/examples/change_streams/with_pipeline.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright 2009-present MongoDB, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstring>
+
+#include <bsoncxx/document/view.hpp>
+#include <bsoncxx/json.hpp>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/client_session.hpp>
+#include <mongocxx/collection.hpp>
+#include <mongocxx/exception/exception.hpp>
+#include <mongocxx/uri.hpp>
+
+#include <examples/api/concern.hh>
+#include <examples/api/db_lock.hh>
+#include <examples/api/runner.hh>
+#include <examples/macros.hh>
+
+namespace {
+
+// [Example]
+void example(mongocxx::client_session session, mongocxx::collection coll) {
+    mongocxx::pipeline pipeline;
+
+    pipeline.match(
+        bsoncxx::from_json(R"({"operationType": "insert", "fullDocument.watched": true})"));
+    // ... other pipeline options.
+
+    mongocxx::change_stream stream = coll.watch(session, pipeline);
+
+    {
+        session.start_transaction(
+            mongocxx::options::transaction{}.max_commit_time_ms(std::chrono::seconds(1)));
+
+        // Observed.
+        EXPECT(coll.insert_one(session, bsoncxx::from_json(R"({"x": 1, "watched": true})")));
+
+        // Not observed (fullDocument mismatch).
+        EXPECT(coll.insert_one(session, bsoncxx::from_json(R"({"x": 2, "watched": false})")));
+
+        // Not observed (operationType mismatch).
+        EXPECT(coll.update_one(session,
+                               bsoncxx::from_json(R"({"x": 2})"),
+                               bsoncxx::from_json(R"({"$set": {"watched": true}})")));
+
+        // Observed.
+        EXPECT(coll.insert_one(session, bsoncxx::from_json(R"({"x": 3, "watched": true})")));
+
+        session.commit_transaction();
+    }
+
+    int count = 0;
+    auto now = [] { return std::chrono::steady_clock::now(); };
+    auto start = now();
+
+    // periodicNoopIntervalSecs: 10 (default)
+    while (count == 0 && now() - start < std::chrono::seconds(10)) {
```

Comment:
```suggestion
    while (count < 2 && now() - start < std::chrono::seconds(10)) {
```

Suggest repeatedly looping until both the first and last inserts are observed. Suggest removing transaction to simplify.

I am not sure if the transaction guarantees the change stream will observe events together (i.e. will all events be returned in one iteration of the `for` loop?). But I expect that is not relevant for this example.